### DISCUSSION TITLE
Token limit

### DISCUSSION THREAD

**Discussion by darkanubis0100** [libraryofcelsus/Aetherius_AI_Assistant#13] [General]
Are there plans to resolve the token limit? Every time I open a chat with the agent, after 3 conversations the token limit appears.

Error communicating with OpenAI: "This model's maximum context length is 4097 tokens. However, your messages resulted in 4181 tokens. Please reduce the length of the messages." - Retrying in 160 seconds...

**Comment by libraryofcelsus** [comment]
> Are there plans to resolve the token limit? Every time I open a chat with the agent, after 3 conversations the token limit appears.
> 
> Error communicating with OpenAI: "This model's maximum context length is 4097 tokens. However, your messages resulted in 4181 tokens. Please reduce the length of the messages." - Retrying in 160 seconds...

Ah sorry, that happens because it's attempting to load to many things from the database.  Quick fix is just to lower the number of entries returned with the top_k modifier.  That issue should get fixed as I'm working on the gui, currently I am trying to narrow in on the maximum entries that can be returned.

**Reply by darkanubis0100** [reply]
I tried downloading the top_k significantly but there is no way, and that limit warning happens much faster if I am working with code.

The ideal solution would obviously be to work with GPT-4, but I've been waiting for almost a year for access and they just ignore.

**Reply by libraryofcelsus** [reply]
Ah, yah the current script is meant for gpt-4.  I will be remaking the 3.5 only version once I finish transferring over all the features to the gui.  Will probably be done in a week or two.

**Reply by darkanubis0100** [reply]
No problems, I'm still seeing what else can be done or should be corrected as time goes on. But I can tell you that this is very useful, GPT really understands what you are talking about thanks to this memory.
